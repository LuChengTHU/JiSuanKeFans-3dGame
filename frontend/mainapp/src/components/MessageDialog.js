/* eslint-disable flowtype/require-valid-file-annotation */

import React from 'react';
import Button from 'material-ui/Button';
import { withStyles } from 'material-ui/styles';
import Dialog, {
  DialogActions,
  DialogContent,
  DialogTitle,
} from 'material-ui/Dialog';
import PropTypes from 'prop-types';

const styles = (/*theme*/) => ({
})

class MessageDialog extends React.Component {
  constructor(props) 
  {
    super(props);
    this.muiName = 'Dialog';
  }

  render() {
    const { onRequestClose, closeText, onRequestConfirm, confirmText, title, ...other } = this.props;
    let confirmButton = null, closeButton = null;
    if (onRequestConfirm)
    {
        confirmButton = <Button id={'confirm'} onClick={onRequestConfirm} color="primary">
                { confirmText }
            </Button>;
    }
    if (onRequestClose)
    {
        closeButton = <Button id={'cancel'} onClick={onRequestClose} color="primary">
                { closeText }
            </Button>;
    }
    return (
        <Dialog {...other} onRequestClose={onRequestClose} >
          <DialogTitle>{ title }</DialogTitle>
          <DialogContent>
            { this.props.children }
          </DialogContent>
          <DialogActions>
            { confirmButton }
            { closeButton }
          </DialogActions>
        </Dialog>
    );
  }
}

MessageDialog.propTypes = {
    title: PropTypes.string.isRequired,
    closeText: PropTypes.string,
    onRequestClose: PropTypes.func,
    confirmText: PropTypes.string,
    onRequestConfirm: PropTypes.func,
};

export default withStyles(styles)(MessageDialog);