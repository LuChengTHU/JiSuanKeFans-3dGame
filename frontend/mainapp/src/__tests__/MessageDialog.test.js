import React, {Component} from 'react';
import MessageDialog from '../components/MessageDialog';
import TestRenderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import {shallow, configure, mount, render} from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import { withStyles, MuiThemeProvider } from 'material-ui/styles';
import createContext from '../styles/createContext';
import toJson from 'enzyme-to-json';
configure({ adapter: new Adapter() });
const context = createContext();
const setup = () => {
  const props = {
    onRequestClose: jest.fn(),
    onRequestConfirm: jest.fn(),
    closeText:'close',
    confirmText:'confirm',
    open:true,
    title:'title'
  }
  const wrapper = shallow(
    <MessageDialog {...props} >
      <div>
        content
      </div>
    </MessageDialog>
  ).dive()
  return {
    props,
    wrapper
  }
}

test('renders correctly && click callback correctly', () => {
  const {wrapper, props} = setup();
  expect(wrapper).toMatchSnapshot();

  wrapper.find('#cancel').simulate('click');
  expect(props.onRequestClose).toHaveBeenCalledTimes(1)
  expect(props.onRequestConfirm).toHaveBeenCalledTimes(0)

  wrapper.find('#confirm').simulate('click');
  expect(props.onRequestClose).toHaveBeenCalledTimes(1)
  expect(props.onRequestConfirm).toHaveBeenCalledTimes(1)
});