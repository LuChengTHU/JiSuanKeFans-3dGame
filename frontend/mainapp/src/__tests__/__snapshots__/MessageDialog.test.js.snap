// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MessageDialog renders correctly && click callback correctly 1`] = `
<withStyles(Dialog)
  onRequestClose={[Function]}
  open={true}
>
  <withStyles(DialogTitle)>
    title
  </withStyles(DialogTitle)>
  <withStyles(DialogContent)>
    <div>
      content
    </div>
  </withStyles(DialogContent)>
  <withStyles(DialogActions)>
    <withStyles(Button)
      color="primary"
      id="confirm"
      onClick={[Function]}
    >
      confirm
    </withStyles(Button)>
    <withStyles(Button)
      color="primary"
      id="cancel"
      onClick={[Function]}
    >
      close
    </withStyles(Button)>
  </withStyles(DialogActions)>
</withStyles(Dialog)>
`;

exports[`components/MessageDialog renders without close correctly 1`] = `
<withStyles(Dialog)
  open={true}
>
  <withStyles(DialogTitle)>
    title
  </withStyles(DialogTitle)>
  <withStyles(DialogContent)>
    <div>
      content
    </div>
  </withStyles(DialogContent)>
  <withStyles(DialogActions)>
    <withStyles(Button)
      color="primary"
      id="confirm"
      onClick={[Function]}
    >
      confirm
    </withStyles(Button)>
  </withStyles(DialogActions)>
</withStyles(Dialog)>
`;

exports[`components/MessageDialog renders without confirm correctly 1`] = `
<withStyles(Dialog)
  onRequestClose={[Function]}
  open={true}
>
  <withStyles(DialogTitle)>
    title
  </withStyles(DialogTitle)>
  <withStyles(DialogContent)>
    <div>
      content
    </div>
  </withStyles(DialogContent)>
  <withStyles(DialogActions)>
    <withStyles(Button)
      color="primary"
      id="cancel"
      onClick={[Function]}
    >
      close
    </withStyles(Button)>
  </withStyles(DialogActions)>
</withStyles(Dialog)>
`;

exports[`components/MessageDialog renders without confirm nor close correctly 1`] = `
<withStyles(Dialog)
  open={true}
>
  <withStyles(DialogTitle)>
    title
  </withStyles(DialogTitle)>
  <withStyles(DialogContent)>
    <div>
      content
    </div>
  </withStyles(DialogContent)>
  <withStyles(DialogActions) />
</withStyles(Dialog)>
`;
